HW0/P5/P5.txt

```diff
@@ -1 +1,168 @@
+
+Answer: In python, it takes infinite employees np.ceil(np.log2(this_many)) seconds to count this_many bags.  
+
+
+
+
+Problem 5
+------------------
+This would add a second between each of the eight steps.  
+
+Answer: It would take them 15 seconds.   
+
+
+
+
+Problem 6
+---------------
+Assuming that I have to hand out each bag to one of 256 cashiers, that is, it takes me 256 seconds just to pass out the bags.  In the meantime, they are passing bags to each other, making additions, and communicating numbers to each other.  I am assuming that anyone needing to make an addition also needs to be holding the bags for the totals they are adding, and that they cannot add, communicate or pass bags, or receive bags simultaneously.  I am also assuming that there is no need to communicate the number of pennies in an unsummed bag, as anyone can read this number off of the original bag, so the only communication that needs to happen is for numbers that have been summed.  
+
+Second1: 
+I pass out my first bag to cashier 1.  
+I have 255 bags left, one cashier is holding one bag.  
+Second 2:
+I pass out the second bag to cashier 2.  
+I have 254 bags left, two cashiers are each holding a bag.  
+Second 3:
+I pass out a third bag to cashier 3.  
+I have 253 bags left.  Cashier 2 hands her bag to cashier 1 and then vanishes.  
+Second 4:
+I pass out a fourth bag.  
+I have 252 bags left.  Cashier 1 adds the first two bags together. Cashier 3 is standing idle.  
+Second 5:
+I pass out a fifth bag to cashier 5.  
+I have 251 bags left.  Cashier 4 hands her bag to cashier 3 and vanishes.  Cashier 1 is idle.  
+Second 6:
+I pass out a sixth bag to cashier 6.  
+I have 250 bags left.  Cashier 3 adds the two bags that she is holding.  Cashier 5 hands her bag to cashier 1 and vanishes.  
+Second 7:
+I pass out a seventh bag to cashier 7.  
+I have 249 bags left.  Cashier 1 sums the bags that she is holding.  Cashier 6 hands her bag to cashier 3 and vanishes.  
+Second 8:
+I pass out an eighth bag to cashier 8.  
+I have 248 bags left.  Cashier 3 sums the bags that she is holding.  Cashier 7 hands her bag to Cashier 1 and vanishes.  
+Second 9:
+I pass out a ninth bag to cashier 9.  
+I have 247 bags left.  Cashier 1 sums the bags that she is holding.  Cashier 8 hands her bag to cashier 3 and vanishes.  
+...
+In this way, cashier 1 and cashier 3 are keeping a running total, and no one is communicating anything.  
+...
+Second 256:
+I pass out the 256th bag to cashier 256.  
+I have no bags left.  Cashier 3 sums the bags that she is holding.  Cashier 255 hands her bag to cashier 1.  
+Second 257:
+Cashier 1 sums the bags that she is holding.  Cashier 3 is idle.  
+Second 258:
+Cashier 3 hands her bags to cashier 1.  
+Second 259:
+Cashier 3 tells cashier 1 her running total.  
+Second 260:
+Cashier 1 adds the last two numbers together.  
+
+Answer: In the first situation described, it takes 260 seconds for the cashiers to verify 256 bags.   
+
```

Comment:
Well, I got the same answers as the two people I reviewed, and they did it without writing out every step, so I guess I was on the right track.